plugins/ShapeLibrary/stroke_io.lua

```diff
@@ -0,0 +1,89 @@
+local _M = {} -- functions to export
+
+-- Function to read and provide formatted stroke data from a shape-file
+function _M.read_strokes_from_file(filepath)
+    if filepath == nil then return end
+    local hasFile, content = pcall(dofile, filepath)
```

Comment:
Not sure if it's really necessary: In theory this file can contain anything, including arbitrary code which gets executed here (not just only returning a table, but doing `os.execute` things).
In another project, I tried to sandbox the environment in which the file is loaded (see https://github.com/atticus-sullivan/cluttealtex/blob/main/src_teal/texrunner/read_cfg.lua#L35-L65). For this purpose we could go even further, denying access to all global variables (including `os`, `debug`, etc) for the lua file loaded (by simply setting a proper `__index` function for meta-table of the `sandbox` table). If we also deny such accesses, we also wouldn't need to do any deep copies)

This makes the assumption, that the file containing the information about the strokes is "untrusted" and might contain "malicious" code (note the "config" could also do stuff like `os.execute = function(...) end`, emplacing hooks for stdlib functions).
Not sure if we need to safeguard against something like that. After all the user can also provide all kinds of "bad" plugins anyhow.

Reply:
Thanks for the code snippet. I will think about it. It looks quite hard to securely prevent any unwanted code execution by a malicious third party, but we could definitely do some hardening.

Reply:
Well actually in case we really only want the config to return a table (so it's just some constants), I think it should be as easy as don't allow any variables from the global lua state (the script shouldn't be able to `require`, `os.execute` etc as this accesses the corresponding global variables `require`, `os` etc).

But still a malicous third party could also adjust the plugin directly :shrug:

In that other project, the idea was less to protect from a malicious third party, but to safeguard from users having typos in the config or accidentally tampering with lua globals which breaks the rest of the program

Reply:
The plugin, when part of the Xournal++ repo, will go into a system directory (e.g. `/usr/share/xournalpp/plugins`) which may not be accessible to a malicious third party.